DT-6297 use latest graphql queries everywhere

vesameskanen opened the pull request:
This PR removes all references to deprecated graphQL fields. Especially:
- Unified vehicleParking is now used instead of separate carPark and bikePark. Lots of duplicate code removed.
- New time types for itineraries adopted

Changes also include component refactoring and misc bug fixes. Lots of repeating code removed.

Itinerary page related components no longer use moment library and the component tree is now collected into a single 'itinerary' folder.

After this PR, linter gives one warning : UI is already using the new PlanConnection.

vesameskanen committed:
feat: use the new vehiclerentalstation schema

vesameskanen committed:
feat: update bike/car park queries to new vehicleParking schema

vesameskanen committed:
feat: pass vehicle park type as prop to StopPageMap

vesameskanen committed:
feat: combine identical containers for car and bike park maps

vesameskanen committed:
feat: combine almost identical containers for car and bike parking content

vesameskanen committed:
fix: rename a container as container

Because it is a container

vesameskanen committed:
feat: update parkRide to use new schema

vesameskanen committed:
chore: expand parkShape

vesameskanen committed:
feat: update vehicle parking page routes

No duplication any more

vesameskanen committed:
fix: remove unused parking id

vesameskanen committed:
fix: update vehicle park references and refactor ItineraryLegs

vesameskanen committed:
fix: some sneaky lurking bike/car park references

vesameskanen committed:
fix: add default prop

vesameskanen committed:
fix: use itinerary start and end instead of startTime and endTime

vesameskanen committed:
chore: remove references to unused and deprecated arrivalTime

vesameskanen committed:
chore: remove vague test

vesameskanen committed:
chore: update arrivalTime to arrival in unit tests

vesameskanen committed:
feat: use the new legTime schema

vesameskanen committed:
fix: refactor Itinerary.js time computations

vesameskanen committed:
fix: use legTime in unit tests

vesameskanen committed:
chore: refactor AlternativeLegsInfo.js

Do not duplicate lots of jsx

vesameskanen committed:
fix: misc bugs from legTime change

vesameskanen committed:
fix: missing index prop for airport related legs

vesameskanen committed:
fix: add missing translation

vesameskanen committed:
fix: WaitLeg time prop

vesameskanen committed:
fix: remove lots of unnecessary repetition

vesameskanen committed:
fix: more test updates

vesameskanen committed:
chore: remove unsupported openinghours temporarily

vesameskanen committed:
chore: update deprecated alerts query

vesameskanen committed:
chore: simpler container name

vesameskanen committed:
fix: use scoped id for park paths

vesameskanen committed:
chore: update graphql schema

optionsome commented on app/util/legUtils.js:
```diff
@@ -3,6 +3,14 @@ import get from 'lodash/get';
+/**
+ * Get time as  milliseconds since the Unix Epoch
+ */
+export function legTime(lt) {
+  const t = lt.estimated?.time || lt.scheduledTime;
+  return Date.parse(t);
+}
```
Quite often we use this for moment. I wonder if it makes any difference to create a method that directly parses the ISO 8601 time to moment.

vesameskanen replied:
I added new util which returns an actual time string in the  form UI frequently uses.

optionsome commented on app/component/AirportCollectLuggageLeg.js:
```diff
@@ -18,7 +18,7 @@ function AirportCollectLuggageLeg(props, { config }) {
       <div className="small-2 columns itinerary-time-column">
         <div className="itinerary-time-column-time">
-          {moment(props.leg.endTime).format('HH:mm')}
+          {moment(props.leg.end).format('HH:mm')}
```
I think you need to use the legTime method here to parse the actual time.

vesameskanen replied:
Right - I grepped 'moment(props.leg', found many of those and fixed them all.

vesameskanen committed:
feat: util for converting legTime to 'hh:mm' string

vesameskanen committed:
fix: use the new legTimeStr util, fix the remaining legTime bugs

vesameskanen committed:
feat: func for epoch ms to 'hh:mm'

vesameskanen committed:
chore: refactor time use in leg components (no moment lib)

vesameskanen committed:
chore: don't use moment object as time parameter

vesameskanen committed:
fix: don't use moment and LocalTime, timeStr is enough

vesameskanen committed:
fix: simplify time usage in Itinerary.js

vesameskanen committed:
fix: remove totally redundant time component

LocalTime does the same time conversion as other time utils and
provides no extra value.

vesameskanen committed:
fix: remove LocalTime from DepartureTime

vesameskanen committed:
fix: remove complicated moment time conversions from Itinerary page and weather queries

vesameskanen committed:
fix: remove moment from weather popup

vesameskanen committed:
chore: remove moment from the rest of itinerary search related places

vesameskanen committed:
chore: add space

vesameskanen committed:
feat: move all itinerary page related components to a dedicated folder

Also:
- Move generic Toggle to componentfolder root
- Remove dead styles
- Refactor some componets

vesameskanen committed:
fix: update unit test paths to components

vesameskanen committed:
fix: some broken paths

vesameskanen committed:
chore: reorder sass imports

vesameskanen committed:
Merge branch 'v3' into DT-6297

vesameskanen committed:
chore: itinerary styles last

vesameskanen committed:
fix: show weather also when temperature is zero

vesameskanen committed:
fix: do not render separator bar into bike/car + public lists

optionsome commented on app/component/ParkOrStationHeader.js:
```diff
@@ -43,8 +46,8 @@ const ParkOrBikeStationHeader = ({ parkOrStation, breakpoint }, { config }) => {
-  const { name, bikeParkId, stationId } = parkOrStation;
-  const parkHeaderId = bikeParkId ? 'bike-park' : 'car_park';
+  const { name, stationId } = parkOrStation;
+  const parkHeaderId = parkType === 'bike' ? 'bike-park' : 'car_park';
```
This is doesn't change this but `-` vs `_` difference is a bit odd.

vesameskanen replied:
Right, translation ids are inconsistent. I fixed that one and removes some unused translations at the same time.

optionsome commented on app/component/itinerary/Itinerary.js:
```diff
@@ -687,35 +690,25 @@ const Itinerary = (
-                  ),
+                  departureTime: firstDeparture
+                    ? legTimeStr(firstDeparture.start)
+                    : 'ggh',
```
I wonder what this `ggh` is.

vesameskanen replied:
Me too. I removed it.

optionsome commented on app/component/itinerary/Legs.js:
```diff
@@ -151,144 +152,42 @@ export default class ItineraryLegs extends React.Component {
       } else if (leg.mode === 'AIRPLANE') {
         legs.push(
           <AirportCheckInLeg
+            index={j - 0.5}
```
What do these fractional indices mean?

vesameskanen replied:
Leg needs a number index prop (it is used as react key). Computation does that.

optionsome commented on app/component/itinerary/Legs.js:
```diff
@@ -151,144 +152,42 @@ export default class ItineraryLegs extends React.Component {
-            focusToLeg={this.focusToLeg(leg)}
-          />,
-        );
+        legs.push(<ViaLeg {...legProps} arrival={startTime} />);
```
This is a bit confusing that the arrival=startTime but it was like this before as well.

vesameskanen replied:
Yes, I guess that is somehow correct.

optionsome commented on app/component/itinerary/Legs.js:
```diff
@@ -312,40 +211,23 @@ export default class ItineraryLegs extends React.Component {
             duration: 0,
-            startTime: 0,
-            endTime: 0,
+            start: leg.start,
+            end: leg.start,
```
Why are these the same?

vesameskanen replied:
Just two valid values. Prop must be given, used to be 0 for both.

optionsome commented on app/component/itinerary/TransitLeg.js:
```diff
@@ -53,8 +57,7 @@ const filterNextLegs = leg => {
   return leg.nextLegs.filter(
-    nextLeg =>
-      moment(nextLeg.startTime).diff(moment(leg.startTime), 'hours') < 12,
+    nextLeg => legTime(nextLeg.start) - legTime(leg.start) < 12 * 3600 * 1000, // 12 hours
```
Can we use the leg duration here or are these legs not the legs returned from OTP (or are there additional legs generated by ui)?

vesameskanen replied:
There is no precomputed duraton as difference is computed from two different legs.

optionsome reviewed (changes requested):
We need to decide what to do with backward compatibility of parking pages as this changes which parking ids are accepted (previously non-feedscoped, now feedscoped).

I think there might be an issue with showing occupancy information at least on itinerary page (the icons don't show up when they do in dev).

Vehicle rental station availability (number of bicycles) is not shown on map.

On parking page, the selected parking is not highlighted on map.

On bicycle parking page, some capacity/availability number is displayed in the content as 0, it wasn't shown at all previously.

Stops near you buttons on frontpage don't show an alert which is visible in the view.

Vehicles are not shown on map in itinerary views.

Duration of a leg is shown as `NaN min` min with interlining trips (can be tested with for example kuopio config with /reitti/Vaahteratie%2024%2C%20Kuopio%3A%3A62.93671883702396%2C27.654132843017578/Korpitie%203%2C%20Kuopio%3A%3A62.85357911448784%2C27.66220092773438/?setTime=true&time=1713886213

vesameskanen committed:
fix: remove mysterious 'ggh'

vesameskanen committed:
fix: inconsistent and unused translations

vesameskanen committed:
fix: one legtime was used in old way

vesameskanen committed:
fix: check existence of firstDeparture

vesameskanen committed:
fix: missing parentheses (broke vehicles on map)

vesameskanen committed:
fix: update alerts query for route mode specific alerts

vesameskanen committed:
fix: zero space rendering in bike Park

vesameskanen committed:
fix: occupancy used deprecated time, remove moment reference

vesameskanen committed:
fix: use simpler static Date.now() method to access current time

vesameskanen committed:
fix: citybike availability on map

vesameskanen committed:
fix: map elements now refer to parks using feedscoped id

vesameskanen commented:
Listed bugs and some more fixed.

vesameskanen committed:
feat: don't repeat slow itinerary search if something is found

vesameskanen commented:
Added also logic for stopping slow searches. 

There is still a question if search iteration should be stopped if response contains an error message. For example, short walk search takes a long time because walking filters away all transit itineraries. Query is repeated 4 times, every time with poor results. Error code is WALKING_BETTER_THAN_TRANSIT.

vesameskanen committed:
chore: remove broken xmllint temporarily

optionsome commented:
We still need to do something about the backwards compatibility of parking pages and the issue with alert not showing on stops near you buttons seems to persists (can be tested with matka config, although might also be that somehow I'm using wrong package version). Otherwise looks like the issues were fixed and found no new issues.

vesameskanen commented:
I seem to get same alert triangles in stopsnearyou buttons in opas.matka.fi and this branch.
Backward compatibility of park links: it is not difficult but requires a bit ugly hsl specific configuration. I wonder if such links exist in reality, and we should take the effort.

vesameskanen committed:
feat: more logic for cancelling itinerary search reps

vesameskanen committed:
fix: error code access in search termination check

vesameskanen committed:
fix: remove redundant count check

vesameskanen committed:
feat: do not search trivial biking trips long time

vesameskanen committed:
fix: refactor minimum co2 search to easily understandable form
